tests/test_mig_install_generateconfs.py

```diff
@@ -79,6 +81,22 @@ def test_option_permanent_freeze(self):
         exit_code = main(test_arguments, _generate_confs=fake_generate_confs, _print=noop)
         self.assertEqual(exit_code, 0)
 
+    def test_option_storage_protocols(self):
+        expected_generated_dir = cleanpath('confs-stdlocal', self,
+                                           ensure_dir=True)
+        with open(os.path.join(expected_generated_dir, "instructions.txt"),
+                  "w"):
+            pass
+        fake_generate_confs = create_fake_generate_confs(
+            dict(destination_dir=expected_generated_dir))
+        test_arguments = ['--storage_protocols', 'proto1,proto2,proto3']
```

Comment:
Minor detail: the proper format for the `--storage_protocols` arg is a space-separated list of protocol names, so this will in effect be treated as a single long name with commas in it. Works for testing but may be slightly misleading, so I'll probably replace commas with space here and in assert below.